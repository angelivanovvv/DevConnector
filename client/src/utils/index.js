import { fromJS } from 'immutable';
import Axios from '../common/api/axios';

export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const transformResponse = response => response ? fromJS(response) : response;
export const transformError = error => error ? fromJS(error) : error;

export const setAuthToken = token => {
  if (token) Axios.defaults.headers.common['x-auth-token'] = token;
  else delete Axios.defaults.headers.common['x-auth-token'];
}