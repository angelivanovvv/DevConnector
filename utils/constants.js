const validationRules = {
  login: {
    email: {
      name: "email",
      value: "Please enter valid email"
    },
    password: {
      name: "password",
      value: "Password is required"
    }
  },
  register: {
    name: {
      name: "name",
      value: "Name is required"
    },
    email: {
      name: "email",
      value: "Please enter valid email"
    },
    password: {
      name: "password",
      value: "Please enter password with 6 or more characters"
    }
  },
  profile: {
    status: {
      name: "status",
      value: "Status is required"
    },
    skills: {
      name: "skills",
      value: "Skills is required"
    }
  },
  experience: {
    title: {
      name: "title",
      value: "Title is required"
    },
    company: {
      name: "company",
      value: "Company is required"
    },
    from: {
      name: "from",
      value: "From date is required"
    }
  },
  education: {
    school: {
      name: "school",
      value: "School is required"
    },
    degree: {
      name: "degree",
      value: "Degree is required"
    },
    fieldofstudy: {
      name: "fieldofstudy",
      value: "Field of study is required"
    },
    from: {
      name: "from",
      value: "From date is required"
    }
  },
  post: {
    text: {
      name: "text",
      value: "Text is required"
    }
  }
};

module.exports = {
  validationRules: validationRules
};
