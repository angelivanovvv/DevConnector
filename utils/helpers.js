const hasErrors = (res, errors) =>
  !errors.isEmpty() ? res.status(400).json({ errors: errors.array() }) : null;

module.exports = {
  hasErrors: hasErrors
};
