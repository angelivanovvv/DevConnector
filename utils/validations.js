const { check } = require("express-validator");
const constants = require("./constants");

const {
  login,
  register,
  profile,
  experience,
  education,
  post
} = constants.validationRules;

const LOGIN = () => [
  check(login.email.name, login.email.value).isEmail(),
  check(login.password.name, login.password.value).exists()
];

const REGISTER = () => [
  check(register.name.name, register.name.value)
    .not()
    .isEmpty(),
  check(register.email.name, register.email.value).isEmail(),
  check(register.password.name, register.password.value).isLength({ min: 6 })
];

const PROFILE = () => [
  check(profile.status.name, profile.status.value)
    .not()
    .isEmpty(),
  check(profile.skills.name, profile.skills.value)
    .not()
    .isEmpty()
];

const EXPERIENCE = () => [
  check(experience.title.name, experience.title.value)
    .not()
    .isEmpty(),
  check(experience.company.name, experience.company.value)
    .not()
    .isEmpty(),
  check(experience.from.name, experience.from.value)
    .not()
    .isEmpty()
];

const EDUCATION = () => [
  check(education.school.name, education.school.value)
    .not()
    .isEmpty(),
  check(education.degree.name, education.degree.value)
    .not()
    .isEmpty(),
  check(education.fieldofstudy.name, education.fieldofstudy.value)
    .not()
    .isEmpty(),
  check(education.from.name, education.from.value)
    .not()
    .isEmpty()
];

const POST = () => [
  check(post.text.name, post.text.value)
    .not()
    .isEmpty()
];

module.exports = {
  login: LOGIN,
  register: REGISTER,
  profile: PROFILE,
  experience: EXPERIENCE,
  education: EDUCATION,
  post: POST
};
