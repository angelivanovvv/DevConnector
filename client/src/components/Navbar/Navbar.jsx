import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import { ROUTES } from "./../../constants/clientRoutes";

const Navbar = ({ isAuthenticated, isLoading, logout }) => {
  const authLinks = (
    <ul className="ang-navbar">
      <li className="nav-item">
        <Link to={ROUTES.PROFILES()}>Developers</Link>
      </li>
      <li className="nav-item">
        <Link to={ROUTES.POSTS()}>Posts</Link>
      </li>
      <li className="ang-nav-item">
        <Link to={ROUTES.DASHBOARD()}>
          <i className="fas fa-user icon-right" />
          <span className="hide-sm">Dashboard</span>
        </Link>
      </li>
      <li className="ang-nav-item">
        <span className="pointer" onClick={logout}>
          <i className="fas fa-sign-out-alt icon-right" />
          <span className="hide-sm">Logout</span>
        </span>
      </li>
    </ul>
  );

  const guestLinks = (
    <ul className="ang-navbar">
      <li className="nav-item">
        <Link to={ROUTES.PROFILES()}>Developers</Link>
      </li>
      <li className="ang-nav-item">
        <Link to={ROUTES.REGISTER()}>Register</Link>
      </li>
      <li className="ang-nav-item">
        <Link to={ROUTES.LOGIN()}>Login</Link>
      </li>
    </ul>
  );

  return (
    <nav data-test="ang-navbar" className="ang-navbar navbar bg-dark">
      <h1 data-test="ang-logo">
        <Link to={ROUTES.LANDING()}>
          <i className="fas fa-code" /> DevConnector
        </Link>
      </h1>
      {!isLoading && (
        <Fragment>{isAuthenticated ? authLinks : guestLinks}</Fragment>
      )}
    </nav>
  );
};

Navbar.propTypes = {
  isAuthenticated: PropTypes.bool,
  isLoading: PropTypes.bool,
  logout: PropTypes.func
};

Navbar.defaultProps = {
  isAuthenticated: false,
  isLoading: false,
  logout: () => {}
};

export default Navbar;
