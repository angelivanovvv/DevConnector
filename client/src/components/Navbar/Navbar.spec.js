import React from "react";
import { shallow } from "enzyme";

import Navbar from "./Navbar-component";
import { findByTestAttr } from "../../../../utils/index";

// Declare general function to provide component which is tested
const setUpComponent = (props = {}) => {
  const component = shallow(<Navbar {...props} />);
  return component;
};

describe("Navbar component", () => {
  let component;

  // Declare component before every single test
  beforeEach(() => {
    component = setUpComponent();
  });
  //test one
  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  //test two
  it("It should render without errors", () => {
    const wrapper = findByTestAttr(component, "ang-navbar");
    expect(wrapper.length).toBe(1);
  });
  //test tree
  it("Should render a logo", () => {
    const logo = findByTestAttr(component, "ang-logo");
    expect(logo.length).toBe(1);
  });
});
