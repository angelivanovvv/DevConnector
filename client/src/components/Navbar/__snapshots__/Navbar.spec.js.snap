// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `ShallowWrapper {}`;
